samples/transform_feedback_separated_2.html

```diff
@@ -0,0 +1,405 @@
+
+            // Toggle source and destination VBO
+            var sourceVBO = particleVBOs[currentSourceIdx];
+            var sourceVAO = particleVAOs[currentSourceIdx];
+
+            var destinationVBO = particleVBOs[(currentSourceIdx + 1) % 2];
+
+            gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, transformFeedback);
+
+            gl.useProgram(programEmit);
+
+            setupVertexAttributes(sourceVAO, sourceVBO);
+
+            // Set transform feedback buffer
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, destinationVBO[Particle.POSITION]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 1, destinationVBO[Particle.VELOCITY]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 2, destinationVBO[Particle.SPAWN_TIME]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 3, destinationVBO[Particle.LIFE_TIME]);
+
+            // Turn off rasterization - we are not drawing
+            gl.enable(gl.RASTERIZER_DISCARD);
+
+            // Set uniforms
+            gl.uniform1f(emitTimeLocation, time);
+            gl.uniform2f(emitAccelerationLocation, 0.0, ACCELERATION);
+
+            // Emit particles using transform feedback
+            gl.beginTransformFeedback(gl.POINTS);
+            gl.drawArrays(gl.POINTS, 0, NUM_PARTICLES);
+            gl.endTransformFeedback();
+
+            // Restore state
+            gl.disable(gl.RASTERIZER_DISCARD);
+            gl.useProgram(null);
+            gl.bindBuffer(gl.ARRAY_BUFFER, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 1, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 2, null);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 3, null);
+            gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+
+            // Ping pong the buffers
+            currentSourceIdx = (currentSourceIdx + 1) % 2;
+        }
+
+        function render() {
+            // Spawn particles
+            emitParticles();
+
+            time = Date.now();
+
+            // Set the viewport
+            gl.viewport(0, 0, canvas.width, canvas.height - 10);
+
+            // Clear color buffer
+            gl.clearColor(0.0, 0.0, 0.0, 1.0);
+            gl.clear(gl.COLOR_BUFFER_BIT);
+
+            setupVertexAttributes(particleVAOs[currentSourceIdx], particleVBOs[currentSourceIdx]);
+
+            gl.useProgram(programs[PROGRAM_DRAW]);
+
+            // Set uniforms
+            gl.uniform1f(drawTimeLocation, time);
+            gl.uniform4f(drawColorLocation, 0.0, 1.0, 1.0, 1.0);
+            gl.uniform2f(drawAccelerationLocation, 0.0, ACCELERATION);
+
+            // Enable blending
+            gl.enable(gl.BLEND);
+            gl.blendFunc(gl.SRC_ALPHA, gl.ONE);
+
+            gl.drawArrays(gl.POINTS, 0, NUM_PARTICLES);
+
+            requestAnimationFrame(render);
+        }
+
+        window.onunload = function() {

```

Comment:
Setting an onunload handler to do cleanup isn't necessary, and will slow down page shutdown in Chrome at least. I recommend simply deleting this code. The WebGL implementation will delete these resources, typically as soon as page navigation occurs.